### DISCUSSION TITLE
Don't manage to create a url like /.well-known/webfinger

### DISCUSSION THREAD

**Discussion by flq** [gatsbyjs/gatsby#36976] [Help]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

I'm trying to add a piece of static json to be served at a specific url such that I can give people a fediverse user handle that points to my domain and it can then direct a questioning client to the right place. For this one needs to serve a specific piece of JSON as is described e.g. in this post: [Mastodon on your domain without hosting a server](https://blog.maartenballiauw.be/post/2022/11/05/mastodon-own-donain-without-hosting-server.html). Long story short, the url needs to be like /.well-known/webfinger

What I've tried: 
1. Added a .well-known directory with webfinger file containing the correct json
    * worked in development, but on production it redirected me to my 404 page
2. Under pages, added ".well-known" directory, underneath "webfinger.js" with the default component serving the json as string
3. With the "createPage" action, created the page at the desired place.

What is sad is, all these approaches **work during develop**, but **none** on production build.

For 2 and 3, I get an exception during build:

```
 ERROR 

Page data from page-data.json for the failed page "/.well-known/webfinger": {
  "componentChunkName": "component---src-components-webfinger-js",
  "path": "/.well-known/webfinger",
  "result": {
    "pageContext": {}
  },
  "staticQueryHashes": []
}

failed Building static HTML for pages - 1.836s

 ERROR #95313 

Building static HTML failed for path "/.well-known/webfinger"

Error: ENOTDIR: not a directory, open '/Users/fquednau/Documents/github/realfiction/public/.well-known/webfinger/index.html'
```

### Reproduction Link

https://github.com/flq/gatsby-well-known-issue.git

### Steps to Reproduce

1. Try to create a file served at /.well-known/webfinger

### Expected Result

The different strategies work as they work during development.

### Actual Result

See exception in the description

### Environment

```shell
System:
    OS: macOS 12.6
    CPU: (8) arm64 Apple M2
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 16.17.0 - /var/folders/d8/r0qgc5q10677q75vjckymyp80000gn/T/yarn--1667691125175-0.037421269093872445/node
    Yarn: 1.22.19 - /var/folders/d8/r0qgc5q10677q75vjckymyp80000gn/T/yarn--1667691125175-0.037421269093872445/yarn
    npm: 8.15.0 - /opt/homebrew/opt/node@16/bin/npm
  Browsers:
    Chrome: 107.0.5304.87
    Safari: 16.0
  npmPackages:
    gatsby: ^4.24.4 => 4.24.4
    gatsby-image: ^3.9.0 => 3.11.0
    gatsby-plugin-feed-mdx: ^1.1.0 => 1.1.0
    gatsby-plugin-google-analytics: ^4.24.0 => 4.24.0
    gatsby-plugin-manifest: ^4.2.0 => 4.17.0
    gatsby-plugin-mdx: ^3.2.0 => 3.17.0
    gatsby-plugin-offline: ^5.24.0 => 5.24.0
    gatsby-plugin-react-helmet: ^5.24.0 => 5.24.0
    gatsby-plugin-sitemap: ^5.24.0 => 5.24.0
    gatsby-remark-prismjs: ^6.24.0 => 6.24.0
    gatsby-source-filesystem: ^4.24.0 => 4.24.0
```

**Comment by LekoArts** [comment]
Put the file into the `static` folder: https://www.gatsbyjs.com/docs/how-to/images-and-media/static-folder/
